### ISSUE TITLE
The microservice node is collapsed and then re-expanded after single click the collapse icon

### ISSUE THREAD

**Issue by v-ruizh** [microsoft/vscode-tye#106]
**OS:** Win10
**Build Version:** 20210628.1

**Repro Steps:**
1. Download the project by the command "git clone https://github.com/pratiksanglikar/tye-demo" -> Open the project in VS Code -> Execute "tye run" for this project.
2. Navigate to Tye view and try to single click the collapse icon of the microservice node.
3. Check whether succeed to collapse the microservice node.

**Expect:**
Succeed to collapse the microservice node.

**Actual:**
The microservice node is collapsed and then re-expanded after single clicking the collapse icon.
![ghf](https://user-images.githubusercontent.com/45614674/123944250-85bac680-d9cf-11eb-8c3f-bda40e7afed9.gif)

**More Info:**
This issue does not reproduce after single clicking other service node in Tye view.

**Comment by philliphoff** [comment]
I found a bug in our process comparison logic which caused the tree view to refresh much more frequently than required, which helped to expose the issue.  This will be fixed in an upcoming PR (so that the tree view will only actually be refreshed when a Tye application changes as intended).

However, there seems to be an issue with VS Code itself that doesn't respect an expanded-by-default node's current collapsed state when refreshed (whereas, a collapsed-by-default node's current expanded state will be respected after a refresh).  I filed this as [vscode#127711](https://github.com/microsoft/vscode/issues/127711).  There may ways to workaround this in the meantime, but I'm not sure that meets the bar for 0.2 (given it should be far less obvious with the aforementioned fix).
